test/issue-28.test.js

```diff
@@ -575,3 +575,29 @@ test('Nested wildcards with parametric and static - 8', t => {
     null
   )
 })
+
+test('Wildcard node with constraints', t => {
+  t.plan(1)
+  const findMyWay = FindMyWay({
+    defaultRoute: (req, res) => {
+      t.fail('we should not be here, the url is: ' + req.url)
+    }
+  })
+
+  findMyWay.on('GET', '*', (req, res, params) => {
+    t.equal(params['*'], '/foo1/foo3')
+  })
+
+  findMyWay.on('GET', '/foo1/*', { constraints: { host: 'fastify.io' } }, (req, res, params) => {
```

Comment:
Could you add a test case that hit this route with constraint?
It is a valuable test that the defaultRoute is not hit in this case too

Reply:
Done